src/conn/mod.rs

```diff
@@ -363,19 +370,18 @@ async fn run(
 
 async fn send_answer(
     socket: &Arc<UdpSocket>,
+    local_ips: &[IpAddr],
     name: &str,
     dst: IpAddr,
     dst_addr: SocketAddr,
 ) -> Result<(), Error> {
     let raw_answer = {
-        let mut msg = Message {
-            header: Header {
-                response: true,
-                authoritative: true,
-                ..Default::default()
-            },
-
-            answers: vec![Resource {
+        let mut answers = vec![];
+        for local_ip in local_ips {
+            if local_ip.is_loopback() && !dst.is_loopback() {
+                continue
```

Comment:
Thanks for the PR @chertov. I do notice issues due to IPV6 interfaces here, we are not checking if the interface address is an IPV6 causing ` IpAddr::V6(_) => return Err(Error::new("Unexpected IpV6 addr".to_owned())),` to be called. 
Also, we are creating answers for all non-loopback interfaces instead of just the interface that we received the question.
Can we instead query for the local interface address of the source IP and send as answer body? something like this
```rust
async fn get_interface_addr_for_ip(addr: impl ToSocketAddrs) -> std::io::Result<SocketAddr> {
    let socket = UdpSocket::bind("0.0.0.0:0").await?;

    socket.connect(addr).await?;

    socket.local_addr()
}

```
`get_interface_addr_for_ip` function can be called in the `start` function, just before `run` is called.

Reply:
Theres currently no IPV6 support. IPV6 support will be added very much later.